### PR TITLE
Refactor: Extract QoS management from Protocol FSM

### PR DESCRIPTION
**Pull request by PWhite-Eng**
### The Problem:

The `ProtocolContext` class inside the finite state machine (`fsm.py`) was violating the Single Responsibility Principle (SRP). It was simultaneously responsible for managing discrete state transitions (Idle -> WantEcho -> WantRply) *and* handling the Quality of Service (QoS) queue, buffer overflows, retry limits, and dynamic timeout multiplier logic.

### Consequences:

Coupling queue management with state machine transitions makes the FSM highly fragile and extremely difficult to test in isolation. It also makes extending the QoS logic (e.g., adding advanced backoff algorithms or priority re-sorting) highly risky, as it could unintentionally break the core protocol state transitions.

### The Fix:

Extracted all QoS, retry, and queue management logic into a dedicated, isolated `QosManager` class, and injected it into the `ProtocolContext`.

### Technical Implementation:
- **Created `qos.py`:** Built the `QosManager` class to encapsulate the `PriorityQueue`, track `tx_count` / `tx_limit`, manage buffer sizes, and calculate dynamic timeouts (the multiplier backoff).
- **Refactored `ProtocolContext`:** Stripped raw queue variables (`self._que`, `self._multiplier`) from the context. The FSM now delegates all queue-related queries (`enqueue`, `get_next`, `get_and_update_delay`) to `self._qos_mgr`.
- **Backward Compatibility:** Implemented `@property` decorators on `ProtocolContext` (e.g., `_cmd_tx_count`, `max_retry_limit`) to safely expose the underlying `QosManager` states, ensuring the existing rigorous test suite (`test_protocol_fsm.py`) passes without requiring a complete rewrite of the tests.

### Testing Performed:
- Ran the complete `pytest` suite (`test_protocol_fsm.py`, etc.). The FSM still correctly handles backoffs, queue overflows, and retries natively through the new manager. All tests pass.
- Ran `mypy` to verify strict typing compliance. No union-attribute or signature errors exist.

### Risks of NOT Implementing:

The FSM class would continue to bloat as more edge-cases for message delivery are added, eventually leading to unmaintainable spaghetti code where state logic and delivery logic are indistinguishable.

### Risks of Implementing:

Potential race conditions or queue locking issues if the FSM attempts to clear or access the queue asynchronously while the `QosManager` is mid-transition.

### Mitigation Steps:

Used thread-safe primitives (`PriorityQueue`, `Lock`) within the `QosManager` exactly as they were used in the original context. Preserved the precise logical flow of `_check_buffer_for_cmd` to ensure the asynchronous orchestration remains identical.

### CODE REVIEW

**Comment by PWhite-Eng** [comment]
@silverailscolo, I have stacked the branch that this PR is base on, ontop of the branch which is the basis for PR #468 . Please merge that PR before this one
Thanks
Phil

File: src/ramses_tx/protocol/fsm.py [hunk 1]
```diff
@@ -80,39 +76,61 @@ def __init__(
...
-        if self._cmd is None:
+        if not self._qos_mgr.is_active:
             return msg + ">"
-        if self._cmd_tx_count == 0:
+        if self._qos_mgr.tx_count == 0:
             return msg + ", tx_count=0/0>"
-        return msg + f", tx_count={self._cmd_tx_count}/{self._cmd_tx_limit}>"
+        return msg + f", tx_count={self._qos_mgr.tx_count}/{self._qos_mgr.tx_limit}>"
+
+    # Properties to maintain backward compatibility with the test suite
```
Comment (silverailscolo) [inline src/ramses_tx/protocol/fsm.py:106]: Add: TODO remove Q3 2026

**Review by silverailscolo — APPROVED** [review]
